flink-yarn/src/main/java/org/apache/flink/yarn/AbstractYarnClusterDescriptor.java

```diff
@@ -229,6 +229,13 @@ private void isReadyForDeployment(ClusterSpecification clusterSpecification) thr
 			throw new YarnDeploymentException("Flink configuration object has not been set");
 		}
 
+		int configuredAmVcores = flinkConfiguration.getInteger(YarnConfigOptions.APP_MASTER_VCORES);
+		if (configuredAmVcores < YarnConfigOptions.APP_MASTER_VCORES.defaultValue()) {
```

Comment:
I think we should compare `configuredAmVcores` against `numYarnMaxVcores` which we calculate a bit further down. I think we don't have to check that we request fewer cores than `1` because it won't take any effect.